mlir/test/Quantum/ScatterTest.mlir

```diff
@@ -117,13 +122,28 @@ func.func public @two_scatter(%arg0: tensor<3xf64>, %arg1: tensor<i64>) -> tenso
     // CHECK-NEXT:   return [[RESTENSOR1]] : tensor<f64>
 
 // CHECK: func.func public @two_scatter(%arg0: tensor<3xf64>, %arg1: tensor<i64>) -> tensor<3xf64>
-    // CHECK:   [[CST:%.+]] = arith.constant dense<3.000000e+00> : tensor<f64>
-    // CHECK:   [[CST0:%.+]] = arith.constant dense<2.000000e+00> : tensor<f64>
-    // CHECK:   [[INPUTS0:%.+]] = tensor.from_elements %extracted_3 : tensor<f64>
-    // CHECK:   [[CALL0:%.+]] = call @__catalyst_update_scatter[[NUMBER0]]([[INPUTS0]], [[CST]]) : (tensor<f64>, tensor<f64>) -> tensor<f64>
-    // CHECK:   [[INPUTS1:%.+]] = tensor.extract %arg0[%10] : tensor<3xf64>
-    // CHECK:   [[INPUTSEXTRACTED1:%.+]] = tensor.from_elements [[INPUTS1]] : tensor<f64>
-    // CHECK:   [[CALL1:%.+]] = call @__catalyst_update_scatter[[NUMBER1]]([[INPUTSEXTRACTED1]], [[CST0]]) : (tensor<f64>, tensor<f64>) -> tensor<f64>  
+    // CHECK:    [[IDX0:%.+]] = index.constant 0
+    // CHECK:    [[IDX1:%.+]] = index.constant 1
+    // CHECK:    [[CST0:%.+]] = arith.constant dense<2.000000e+00> : tensor<f64>
+    // CHECK:    [[CST1:%.+]] = arith.constant dense<3.000000e+00> : tensor<f64>
+
+    // CHECK:    [[SCF0:%.+]] = scf.for %arg2 = [[IDX0]] to [[IDX1]] step [[IDX1]] iter_args(%arg3 = %arg0) -> (tensor<3xf64>)
+    // CHECK:    [[INDEX:%.+]] = arith.index_cast %13 : i32 to index
```

Comment:
Same here, `%13` needs to be removed. If we don't want to track where it came from it can be replaced with `{{%.*}}`, otherwise it should be replaced with a matched variable name.

Reply:
Done :+1: